src/Api/Validator.php

```diff
@@ -249,7 +249,20 @@ private function checkCanString($value)
 
     private function checkAssociativeArray($value)
     {
-        if (!is_array($value) || isset($value[0])) {
+        $assoc = false;
+
+        if (is_array($value)) {
+            $i   = 0;
```

Comment:
More style nitpicks:
* Can the variable be more descriptive, like `$expectedIndex`? 
* Can you remove the whitepsace after the variable?